gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/resourcename/ResourceNameHelperClassComposer.java

```diff
@@ -160,8 +160,26 @@ private static List<AnnotationNode> createClassAnnotations() {
             .build());
   }
 
-  private static List<TypeNode> createImplementsTypes() {
-    return Arrays.asList(FIXED_TYPESTORE.get("ResourceName"));
+  /**
+   * Builds a list of types to be implemented. If the type has the same name then we use its full
+   * name
+   *
+   * @param implementingClassName class that is implementing the resulting list. Collisions are
+   *     checked
+   * @return
+   */
+  private static List<TypeNode> createImplementsTypes(String implementingClassName) {
+    List<TypeNode> preprocessed = Arrays.asList(FIXED_TYPESTORE.get("ResourceName"));
+    List<TypeNode> result = new ArrayList<>();
+    for (TypeNode typeNode : preprocessed) {
+      if (!typeNode.isPrimitiveType()
+          && typeNode.reference().name().compareTo(implementingClassName) == 0) {
+        result.add(FIXED_TYPESTORE.getWithFullName(typeNode.reference().name()));
```

Comment:
I know you may want to create a genetic `getWithFullName` method so that it could helpful in the future, but since this is a very specific issue, and we already hardcode the String `ResourceName` in the beginning of the method, I think it might be easier to just fix this issue.

Reply:
Makes sense. I just moved the deduplication logic into `createImplementsTypes`